src/widgets/range-input/range-input.tsx

```diff
@@ -0,0 +1,229 @@
+  /**
+   * The label of the submit button
+   * @default "Go"
+   */
+  submitText: Template;
+};
+
+export type RangeInputCSSClasses = {
+  /**
+   * CSS class to add to the root element.
+   */
+  root: string | string[];
+  /**
+   * CSS class to add to the root element when there's no refinements.
+   */
+  noRefinement: string | string[];
+  /**
+   * CSS class to add to the form element.
+   */
+  form: string | string[];
+  /**
+   * CSS class to add to the label element.
+   */
+  label: string | string[];
+  /**
+   * CSS class to add to the input element.
+   */
+  input: string | string[];
+  /**
+   * CSS class to add to the min input element.
+   */
+  inputMin: string | string[];
+  /**
+   * CSS class to add to the max input element.
+   */
+  separator: string | string[];
+  /**
+   * CSS class to add to the separator of the form.
+   */
+  inputMax: string | string[];
+  /**
+   * CSS class to add to the submit button of the form.
+   */
+  submit: string | string[];
+};
+
+export type RangeInputWidgetParams = {
+  /**
+   * Valid CSS Selector as a string or DOMElement.
+   */
+  container: string | HTMLElement;
+  /**
+   * Name of the attribute for faceting.
+   */
+  attribute: string;
+  /**
+   * Minimal slider value, default to automatically computed from the result set.
+   */
+  min?: number;
+  /**
+   * Maximal slider value, defaults to automatically computed from the result set.
+   */
+  max?: number;
+  /**
+   * Number of digits after decimal point to use.
+   * @default 0
+   */
+  precision?: number;
+  /**
+   * Labels to use for the widget.
+   */
+  templates?: Partial<RangeInputTemplates>;
+  /**
+   * CSS classes to add.
+   */
+  cssClasses?: Partial<RangeInputCSSClasses>;
+};
+
+const renderer = ({
+  containerNode,
+  cssClasses,
+  renderState,
+  templates,
+}): Renderer<RangeRenderState, Partial<RangeInputWidgetParams>> => (
+  { refine, range, start, widgetParams, instantSearchInstance },
+  isFirstRendering
+) => {
+  if (isFirstRendering) {
+    renderState.templateProps = prepareTemplateProps({
+      defaultTemplates,
+      templatesConfig: instantSearchInstance.templatesConfig,
+      templates,
+    });
+    return;
+  }
+
+  const { min: rangeMin, max: rangeMax } = range;
+  const [minValue, maxValue] = start;
+
+  const step = 1 / Math.pow(10, widgetParams.precision || 0);
+
+  const values = {
+    min: minValue !== -Infinity && minValue !== rangeMin ? minValue : undefined,
+    max: maxValue !== Infinity && maxValue !== rangeMax ? maxValue : undefined,
+  };
+
+  render(
+    <RangeInput
+      min={rangeMin}
+      max={rangeMax}
+      step={step}
+      values={values}
+      cssClasses={cssClasses}
+      refine={refine}
+      templateProps={renderState.templateProps}
+    />,
+    containerNode
+  );
+};
+
+export type RangeInputWidget = WidgetFactory<
+  Omit<RangeWidgetDescription, '$$type'> & {
+    $$widgetType: 'ais.rangeInput';
+    $$type: 'ais.rangeInput';
+  },
```

Comment:
good call here, in a later version we can leave $$type as range, it originally wasn't changed since the impact on https://github.com/algolia/instantsearch.js/blob/master/src/lib/utils/checkIndexUiState.ts was too big to use widgetType instead of type.